### ISSUE TITLE
Crash on launch: TypeError: Bitmap.FromRGBA(): argument 1 has unexpected type 'float'

### ISSUE THREAD

**Issue by NickGeek** [hhannine/superpaper#103]
**Describe the bug**
Running Python 3.10.1, I get the following error:
```
Traceback (most recent call last):
  File "/usr/bin/superpaper", line 33, in <module>
    sys.exit(load_entry_point('superpaper==2.1.0', 'console_scripts', 'superpaper')())
  File "/usr/bin/superpaper", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 919, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 518, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for superpaper
```

**To Reproduce**
1. Install Python 3.10.1
2. Start superpaper

**Expected behavior**
The application should start.

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Desktop Environment on Linux: i3wm
 - Superpaper package & version: AUR 2.1.0-1

**Comment by NickGeek** [comment]
The AppImage also doesn't work, but for a different reason:
```
feh: /tmp/_MEIGHqHwh/libz.so.1: version `ZLIB_1.2.9' not found (required by /usr/lib/libpng16.so.16)
```

**Comment by NickGeek** [comment]
I seemed to solve this problem by manually updating deps with pypy, looks like an issue with the AUR package and not superpaper itself.

**Comment by NickGeek** [comment]
Most of the app works except for the main config page:
```
Exception ignored in thread started by: <bound method CallSerializer.call_functions of <system_hotkey.util.CallSerializer object at 0x7fe16cff0220>>
Traceback (most recent call last):
  File "/home/nick/.local/lib/python3.10/site-packages/system_hotkey/util.py", line 37, in call_functions
    func(*args, **kwargs)
  File "/home/nick/.local/lib/python3.10/site-packages/system_hotkey/system_hotkey.py", line 284, in register
    assert isinstance(hotkey, collections.Iterable) and type(hotkey) not in (str, bytes)
AttributeError: module 'collections' has no attribute 'Iterable'
Traceback (most recent call last):
  File "/home/nick/.local/lib/python3.10/site-packages/superpaper/tray.py", line 60, in <lambda>
    menu.Bind(wx.EVT_MENU, lambda event: func(event, *args), id=item.GetId())
  File "/home/nick/.local/lib/python3.10/site-packages/superpaper/tray.py", line 299, in configure_wallpapers
    config_frame = ConfigFrame(self)
  File "/home/nick/.local/lib/python3.10/site-packages/superpaper/gui.py", line 31, in __init__
    config_panel = WallpaperSettingsPanel(self, parent_tray_obj)
  File "/home/nick/.local/lib/python3.10/site-packages/superpaper/gui.py", line 74, in __init__
    self.wpprev_pnl = WallpaperPreviewPanel(self, self.display_sys)
  File "/home/nick/.local/lib/python3.10/site-packages/superpaper/gui.py", line 1363, in __init__
    self.draw_displays()
  File "/home/nick/.local/lib/python3.10/site-packages/superpaper/gui.py", line 1381, in draw_displays
    bmp_canv = wx.Bitmap.FromRGBA(self.dtop_canvas_relsz[0], self.dtop_canvas_relsz[1], red=0, green=0, blue=0, alpha=255)
TypeError: Bitmap.FromRGBA(): argument 1 has unexpected type 'float'
```

**Comment by TimeLincoln** [comment]
Experiencing the same issue on Manjaro using the AUR package.  Updated screeninfo with pip.

`10:11:49: Warning: Mismatch between the program and library build versions detected.
The library used 3.0 (wchar_t,compiler with C++ ABI 1014,wx containers,compatible with 2.8),
and wxPython used 3.0 (wchar_t,compiler with C++ ABI 1016,wx containers,compatible with 2.8).

(superpaper:53801): Gtk-CRITICAL **: 10:11:50.184: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/superpaper/tray.py", line 472, in OnInit
    TaskBarIcon(frame)
  File "/usr/lib/python3.10/site-packages/superpaper/tray.py", line 115, in __init__
    config_frame = ConfigFrame(self)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 31, in __init__
    config_panel = WallpaperSettingsPanel(self, parent_tray_obj)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 74, in __init__
    self.wpprev_pnl = WallpaperPreviewPanel(self, self.display_sys)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 1363, in __init__
    self.draw_displays()
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 1381, in draw_displays
    bmp_canv = wx.Bitmap.FromRGBA(self.dtop_canvas_relsz[0], self.dtop_canvas_relsz[1], red=0, green=0, blue=0, alpha=255)
TypeError: Bitmap.FromRGBA(): argument 1 has unexpected type 'float'
OnInit returned false, exiting...
`

**Comment by adelphi23** [comment]
This is what I have been looking for ages. Unfortunately, it crashes on a clean Fedora 35 install. I am not sure how related to this specific issue it is, but my system does run Python 3.10 so I assume it is related.

└─> superpaper
/home/pc/.local/lib/python3.10/site-packages
/home/pc/.config/superpaper
/home/pc/.config/superpaper/profiles
Exception ignored in thread started by: <bound method CallSerializer.call_functions of <system_hotkey.util.CallSerializer object at 0x7ff66d1918d0>>
Traceback (most recent call last):
  File "/home/pc/.local/lib/python3.10/site-packages/system_hotkey/util.py", line 37, in call_functions
    func(*args, **kwargs)
  File "/home/pc/.local/lib/python3.10/site-packages/system_hotkey/system_hotkey.py", line 284, in register
    assert isinstance(hotkey, collections.Iterable) and type(hotkey) not in (str, bytes)
AttributeError: module 'collections' has no attribute 'Iterable'

(superpaper:8961): Gtk-WARNING **: 12:18:52.628: Theme directory 64/devices of theme Gruvbox has no size field

Traceback (most recent call last):
  File "/home/pc/.local/lib/python3.10/site-packages/superpaper/tray.py", line 472, in OnInit
    TaskBarIcon(frame)
  File "/home/pc/.local/lib/python3.10/site-packages/superpaper/tray.py", line 115, in __init__
    config_frame = ConfigFrame(self)
  File "/home/pc/.local/lib/python3.10/site-packages/superpaper/gui.py", line 31, in __init__
    config_panel = WallpaperSettingsPanel(self, parent_tray_obj)
  File "/home/pc/.local/lib/python3.10/site-packages/superpaper/gui.py", line 74, in __init__
    self.wpprev_pnl = WallpaperPreviewPanel(self, self.display_sys)
  File "/home/pc/.local/lib/python3.10/site-packages/superpaper/gui.py", line 1363, in __init__
    self.draw_displays()
  File "/home/pc/.local/lib/python3.10/site-packages/superpaper/gui.py", line 1381, in draw_displays
    bmp_canv = wx.Bitmap.FromRGBA(self.dtop_canvas_relsz[0], self.dtop_canvas_relsz[1], red=0, green=0, blue=0, alpha=255)
TypeError: Bitmap.FromRGBA(): argument 1 has unexpected type 'float'
OnInit returned false, exiting...

**Comment by ABelliqueux** [comment]
Same error on Manjaro Linux with python 3.10, superpaper 2.1.0-1  from AUR. Same exact error message on launch.

FWIW, here are the dependencies listed in the package, and corresponding libraries versions currently  on my system : 

```
Depends On      :
                  python-pillow>=7.0
                  python-numpy
                  python-wxpython
                  python-xpybutil
                  python-screeninfo>=0.6.1
                  python-system_hotkey>=1.0.2+10+9541587
```
Got : 
```
python-pillow 8.4.0-3
python-numpy 1.21.5-1 
python-wxpython 1:4.0.7.2-3 
python-xpybutil 0.0.6-4
python-screeninfo 0.7-1
python-system_hotkey 1.0.2+12+37073bb-2

I tried tracking down every occurence of `FromRGBA()` in `gui.py` and force the type of the two first arguments to `int()` but to no avail.

Let me know if I can provide more useful information.

**Comment by hhannine** [comment]
This is really perplexing me since I have not reproduced this issue on Manjaro with Python 3.10. 

Can someone run Superpaper with the debugging option '-d' and post the debug logs here?

**Comment by adelphi23** [comment]
> This is really perplexing me since I have not reproduced this issue on Manjaro with Python 3.10.
> 
> Can someone run Superpaper with the debugging option '-d' and post the debug logs here?

Sure!
My configuration is:
- Endeavour OS, 5.16/5-arch1-1, AMD Ryzen Threadripper 1950X, NVIDIA GeForce GTX 1080, 32Gb RAM

Here is the output:
 pc@PCDesktop-EOS: ~ via  16GiB/31GiB | 4GiB/66GiB with /bin/bash 🕙[Tue Feb 08 2022  2:57PM (EST-0500)][2:days up]
┌──────────────────────────────────────────────
└─> superpaper -d
/usr/lib/python3.10/site-packages
/home/pc/.config/superpaper
/home/pc/.config/superpaper/profiles
None
None
None
None
None
None
True
use_hotkeys: True
hk_binding_next: ('control', 'super', 'w')
hk_binding_pause: ('control', 'super', 'shift', 'p')
get_display_data output: NUM_DISPLAYS = 2, RES_ARR = [(2560, 1440), (1440, 2560)], OFF_ARR = [(0, 640), (2560, 0)]
Display(resolution=(2560, 1440), digital_offset=(0, 640), phys_size_mm=(597, 336), detected_phys_size_mm=(597, 336), ppi=108.91792294807368, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='DP-4')
Display(resolution=(1440, 2560), digital_offset=(2560, 0), phys_size_mm=(336, 597), detected_phys_size_mm=(336, 597), ppi=108.85714285714285, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='DP-2')
load_system: archive_file not found: /home/pc/.config/superpaper/display_systems.dat
START Listing profiles for menu.
hkBinding: ('control', 'super', 'shift', 'h')
Listed profile: example
hkBinding: ('control', 'super', 'shift', 'g')
Listed profile: example_multi
END Listing profiles for menu.
No previous profile was found.
hkBinding: ('control', 'super', 'shift', 'h')
Listed profile: example
hkBinding: ('control', 'super', 'shift', 'g')
Listed profile: example_multi
Registering binding:                                 ('control', 'super', 'shift', 'h') for profile: example
Registering binding:                                 ('control', 'super', 'shift', 'g') for profile: example_multi
use_hotkeys: True
hk_binding_next: ('control', 'super', 'w')
hk_binding_pause: ('control', 'super', 'shift', 'p')
get_display_data output: NUM_DISPLAYS = 2, RES_ARR = [(2560, 1440), (1440, 2560)], OFF_ARR = [(0, 640), (2560, 0)]
Display(resolution=(2560, 1440), digital_offset=(0, 640), phys_size_mm=(597, 336), detected_phys_size_mm=(597, 336), ppi=108.91792294807368, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='DP-4')
Display(resolution=(1440, 2560), digital_offset=(2560, 0), phys_size_mm=(336, 597), detected_phys_size_mm=(336, 597), ppi=108.85714285714285, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='DP-2')
load_system: archive_file not found: /home/pc/.config/superpaper/display_systems.dat
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/superpaper/tray.py", line 472, in OnInit
    TaskBarIcon(frame)
  File "/usr/lib/python3.10/site-packages/superpaper/tray.py", line 115, in __init__
    config_frame = ConfigFrame(self)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 31, in __init__
    config_panel = WallpaperSettingsPanel(self, parent_tray_obj)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 74, in __init__
    self.wpprev_pnl = WallpaperPreviewPanel(self, self.display_sys)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 1363, in __init__
    self.draw_displays()
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 1381, in draw_displays
    bmp_canv = wx.Bitmap.FromRGBA(self.dtop_canvas_relsz[0], self.dtop_canvas_relsz[1], red=0, green=0, blue=0, alpha=255)
TypeError: Bitmap.FromRGBA(): argument 1 has unexpected type 'float'
OnInit returned false, exiting...

**Comment by ABelliqueux** [comment]
Here is mine : 

```bash
$ superpaper -d
/usr/lib/python3.10/site-packages
/home/user/.config/superpaper
/home/user/.config/superpaper/profiles
11:59:02: Warning: Mismatch between the program and library build versions detected.
The library used 3.0 (wchar_t,compiler with C++ ABI 1014,wx containers,compatible with 2.8),
and wxPython used 3.0 (wchar_t,compiler with C++ ABI 1016,wx containers,compatible with 2.8).
None
None
None
None
None
None
True
use_hotkeys: True
hk_binding_next: ('control', 'super', 'w')
hk_binding_pause: ('control', 'super', 'shift', 'p')
get_display_data output: NUM_DISPLAYS = 2, RES_ARR = [(900, 1440), (1920, 1080)], OFF_ARR = [(0, 0), (900, 360)]
Display(resolution=(900, 1440), digital_offset=(0, 0), phys_size_mm=(256, 410), detected_phys_size_mm=(256, 410), ppi=89.296875, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='HDMI-1-0')
Display(resolution=(1920, 1080), digital_offset=(900, 360), phys_size_mm=(344, 194), detected_phys_size_mm=(344, 194), ppi=141.7674418604651, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='eDP-1')
load_system: archive_file not found: /home/user/.config/superpaper/display_systems.dat
START Listing profiles for menu.
hkBinding: ('control', 'super', 'shift', 'h')
Listed profile: example
hkBinding: ('control', 'super', 'shift', 'g')
Listed profile: example_multi
END Listing profiles for menu.
No previous profile was found.
hkBinding: ('control', 'super', 'shift', 'h')
Listed profile: example
hkBinding: ('control', 'super', 'shift', 'g')
Listed profile: example_multi
Registering binding:                                 ('control', 'super', 'shift', 'h') for profile: example
Registering binding:                                 ('control', 'super', 'shift', 'g') for profile: example_multi
use_hotkeys: True
hk_binding_next: ('control', 'super', 'w')
hk_binding_pause: ('control', 'super', 'shift', 'p')
get_display_data output: NUM_DISPLAYS = 2, RES_ARR = [(900, 1440), (1920, 1080)], OFF_ARR = [(0, 0), (900, 360)]
Display(resolution=(900, 1440), digital_offset=(0, 0), phys_size_mm=(256, 410), detected_phys_size_mm=(256, 410), ppi=89.296875, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='HDMI-1-0')
Display(resolution=(1920, 1080), digital_offset=(900, 360), phys_size_mm=(344, 194), detected_phys_size_mm=(344, 194), ppi=141.7674418604651, ppi_norm_resolution=None, ppi_norm_offset=None, ppi_norm_bezels=(0, 0), perspective_angles=(0, 0), name='eDP-1')
load_system: archive_file not found: /home/user/.config/superpaper/display_systems.dat
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/superpaper/tray.py", line 472, in OnInit
    TaskBarIcon(frame)
  File "/usr/lib/python3.10/site-packages/superpaper/tray.py", line 115, in __init__
    config_frame = ConfigFrame(self)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 31, in __init__
    config_panel = WallpaperSettingsPanel(self, parent_tray_obj)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 74, in __init__
    self.wpprev_pnl = WallpaperPreviewPanel(self, self.display_sys)
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 1363, in __init__
    self.draw_displays()
  File "/usr/lib/python3.10/site-packages/superpaper/gui.py", line 1381, in draw_displays
    bmp_canv = wx.Bitmap.FromRGBA(self.dtop_canvas_relsz[0], self.dtop_canvas_relsz[1], red=0, green=0, blue=0, alpha=255)
TypeError: Bitmap.FromRGBA(): argument 1 has unexpected type 'float'
OnInit returned false, exiting...
```

**Comment by hhannine** [comment]
Thanks for the logs. I have now reproduced this issue as well and unfortunately it looks like that some changes in wxpython 4 have broken the GUI fairly badly. I was able to correct the above FromRGBA errors to make the GUI open but the wallpaper preview seems to be broken fairly badly and in unexpected ways. Cannot say right now how big of a deal it is going to be to fix.

**Comment by hhannine** [comment]
This was luckily largely resolved by #104.

**Comment by hhannine** [comment]
This has now been largely resolved and will be fixed in the next release. Master branch is mostly working if a little buggy for those willing to run that.
